tests/integration/security/citadel/namespace_targeting_test.go

```diff
@@ -0,0 +1,86 @@
+package citadel
+
+import (
+	"fmt"
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	v1 "k8s.io/api/core/v1"
+	mv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/citadel"
+)
+
+const (
+	testNsPrefix       = "targeting-test"
+	serviceAccountName = "sample-service-account"
+)
+
+// TestNamespaceTargeting verifies that Citadel respects the designated namespace targeting labels.
+// current implementation is closely tied with Kubernetes concepts, but this is difficult to avoid
+// until Citadel itself is less closely tied with Kubernetes concepts
+func TestNamespaceTargeting(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+
+		citadel.NewOrFail(t, ctx, citadel.Config{Istio: ist})
+		env := ctx.Environment().(*kube.Environment)
+
+		// Label combinations
+		type citadelLabels map[string]string
+		testCases := map[string]struct {
+			labels   citadelLabels
+			targeted bool
+		}{
+			"override label set to false, env label matches => should not generate secret": {
+				labels: citadelLabels{
+					"ca.istio.io/override": "false",
+				},
+				targeted: false,
+			},
+			"no override label present, env label matches => should generate secret": {
+				labels: citadelLabels{
+					"ca.istio.io/env": "istio-system", // how can we get this from config?
+				},
+				targeted: true,
+			},
+			"no override label present, no env label present, default is active => should generate secret": {
+				labels:   citadelLabels{},
+				targeted: true,
```

Comment:
What if "ca.istio.io/env": "istio-system", and "ca.istio.io/override": "false" both exist, can we add test to cover that? It is better to cover more cases here.

Reply:
Added a few more cases.

There is also a third configuration option (`enableNamespacesByDefault`), which is toggled through `values.yaml` rather than a namespace label-- not sure how I'd be able to test for this without tearing down/ creating multiple Citadel instances over the course of the test.